aggregator/src/main/java/com/redhat/cloud/notifications/DailyEmailAggregationJob.java

```diff
@@ -34,48 +37,69 @@
-        LocalDateTime now = LocalDateTime.now(UTC);
-        List<AggregationCommand> aggregationCommands = processAggregateEmails(now);
-
-        List<CompletableFuture<Void>> futures = new ArrayList<>();
-        for (AggregationCommand aggregationCommand : aggregationCommands) {
-            try {
-                final String payload = objectMapper.writeValueAsString(aggregationCommand);
-                futures.add(emitter.send(payload).toCompletableFuture());
-            } catch (JsonProcessingException e) {
-                LOG.warn("Could not transform AggregationCommand to JSON object.", e);
+        CollectorRegistry registry = new CollectorRegistry();
+        Gauge duration = Gauge
+                .build()
+                .name("aggregator_job_duration_seconds")
+                .help("Duration of the aggregator job in seconds.")
+                .register(registry);
+        Gauge.Timer durationTimer = duration.startTimer();
+
+        try {
+            LocalDateTime now = LocalDateTime.now(UTC);
+
+            List<AggregationCommand> aggregationCommands = processAggregateEmails(now, registry);
+            List<CompletableFuture<Void>> futures = new ArrayList<>();
+            for (AggregationCommand aggregationCommand : aggregationCommands) {
+                try {
+                    final String payload = objectMapper.writeValueAsString(aggregationCommand);
+                    futures.add(emitter.send(payload).toCompletableFuture());
+                } catch (JsonProcessingException e) {
+                    LOG.warn("Could not transform AggregationCommand to JSON object.", e);
+                }
+
+                // resolve completable futures so the Quarkus main thread doesn't stop before everything has been sent
+                try {
+                    CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).get();
+                } catch (InterruptedException | ExecutionException ie) {
+                    LOG.error("Writing AggregationCommands failed", ie);
+                }
             }
 
-            // resolve completable futures so the Quarkus main thread doesn't stop before everything has been sent
+            emailAggregationResources.updateLastCronJobRun(now);
+
+            Gauge lastSuccess = Gauge
+                    .build()
+                    .name("aggregator_job_last_success")
+                    .help("Last time the aggregator job succeeded.")
+                    .register(registry);
+            lastSuccess.setToCurrentTime();
+        } finally {
+            durationTimer.setDuration();
+            PushGateway pg = new PushGateway(prometheusPushGatewayUrl);
             try {
-                CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).get();
-            } catch (InterruptedException | ExecutionException ie) {
-                LOG.error("Writing AggregationCommands failed", ie);
+                pg.pushAdd(registry, "aggregator_job");
```

Comment:
Will `aggregator_job` be displayed somewhere?

Reply:
Yes, in the Pushgateway of Prometheus.

Reply:
that's the name of the job displayed in the pushgateay ui:
![Screenshot from 2021-12-13 08-38-58](https://user-images.githubusercontent.com/8372856/145771349-ab162bfd-8df6-4022-9886-a2817d5609bf.png)